README.md

```diff
@@ -1,15 +1,120 @@
-# Igor
-[![Build Status](https://api.travis-ci.org/spinnaker/igor.svg?branch=master)](https://travis-ci.org/spinnaker/igor)
+Igor is a service that provides a single point of integration with CI and SCM services for Spinnaker.
 
-Igor provides a single point of integration with Jenkins, Travis and Git repositories ( BitBucket, Stash, and Github ) within Spinnaker.
+# Common Polling Architecture
+
+Igor runs a number of pollers that all share the same common architecture. At a high level, they all:
+
+- periodically get a list of items from an external resource (e.g. builds on a Jenkins master)
+- compare that list against their own persisted cache of items (the difference is called _delta size_)
+- send an `echo` event for each new item
+- caches the new list of items
+
+Features:
+
+- *locking*: pollers can optionally acquire a distributed lock in the storage system before attempting to complete a polling cycle. This makes it possible to run Igor in a high-availability configuration and scale it horizontally.
+- *safe guards*: abnormally large delta sizes can indicate a problem (e.g. lost or corrupt cache data) and cause downstream issues. If a polling cycle results in a delta size above the threshold
```

Comment:
i am sure you wanted to say something else after `above the threshold`

Reply:
Also, maybe worth mentioning the `admin` EP to bypass this safeguard?
`admin/pollers/fastforward/{monitorName}`

Reply:
Haha good catch! I was probably actually wondering what does happen we're above the threshold and then got sidetracked, refrigerator